packages/common/src/ProviderUtils.js

```diff
@@ -0,0 +1,82 @@
+// This client exports a web3 instance which mimics that created by the default truffle config. It exclusively uses
+// web3 providers to take advantage of the flexibility of providing custom configs to tailor the desired options. The network
+// syntax mimics that of the main UMA Truffle implementation to make this backwards compatible.
+
+const Web3 = require("web3");
+const { getTruffleConfig, nodeUrl } = require("./TruffleConfig");
+const argv = require("minimist")(process.argv.slice(), { string: ["network"] });
+const Url = require("url");
+
+// Set web3 to null
+let web3 = null;
+
+function createBasicProvider(url) {
+  const protocol = Url.parse(url).protocol;
+
+  // 10 second timeout
+  const timeout = 10000;
+
+  if (protocol.startsWith("ws")) {
+    // Websocket
+    const websocketOptions = {
+      timeout, // ms
+      clientConfig: {
+        maxReceivedFrameSize: 100000000, // Useful if requests result are large bytes - default: 1MiB
+        maxReceivedMessageSize: 100000000 // bytes - default: 8MiB
+      },
+      reconnect: {
+        auto: true, // Enable auto reconnection
+        delay: 5000, // ms
+        maxAttempts: 10,
+        onTimeout: false
+      }
+    };
+
+    // Create websocket web3 provider. This contains the re-try logic on failed/timeout connections.
+    return new Web3.providers.WebsocketProvider(url, websocketOptions);
+  } else {
+    // If it isn't websocket, assume http.
+    const httpOptions = {
+      timeout
+    };
+
+    return new Web3.providers.HttpProvider(url, httpOptions);
+  }
+}
+
+/**
+ * @notice Gets a web3 instance based on the network argument using the truffle config in this package.
+ */
+function getWeb3() {
+  // If a web3 instance has already been initialized, return it.
+  if (web3) {
+    return web3;
+  }
+
+  // Create basic web3 provider with no wallet connection based on the url alone.
+  const basicProvider = createBasicProvider(nodeUrl);
+
+  // Use the basic provider to create a provider with an unlocked wallet. This piggybacks off the UMA common TruffleConfig
+  // implementing all networks & wallet types. EG: mainnet_mnemonic, kovan_gckms. Errors if no argv.network.
+  const providerWithWallet = getTruffleConfig().networks[argv.network].provider(basicProvider);
```

Comment:
have you tested this on the `test` network with `node`? so something like `node ./packages/monitors/index.js --network test`. I'm pretty sure this wont work unless your other PR [1928](https://github.com/UMAprotocol/protocol/pull/1928) is in first, right?

Reply:
That's correct. After I sent this out for review, I realized there were some test cases that aren't going to work until #1928 is merged. It may make sense to just let #1928 get merged first to simplify things.

Reply:
Done.